### PR TITLE
temporary deployment scheme featuring new/old data.nav.lz4 swap

### PR DESCRIPTION
**Pull request by francois-vincent**

### CODE REVIEW

File: fabfile/component/kraken.py [hunk 1]
```diff
@@ -156,6 +157,22 @@ def test_all_krakens(wait=False):
...
+def swap_all_data_nav():
+    for instance in env.instances.values():
+        swap_data_nav(instance)
+
+def swap_data_nav(instance):
+    """ swap old/new data.nav.lz4, only if new is still in temp directory
+    """
+    temp_target = instance.temp_target_lz4_file
+    plain_target = instance.plain_target_lz4_file
+    if os.access(temp_target, os.W_OK) and os.access(plain_target, os.W_OK) and \
```
Comment (antoine-de) [inline fabfile/component/kraken.py]: in fabtool there is fabtools.files.getmtime
and is_file

File: fabfile/component/kraken.py [hunk 2]
```diff
@@ -156,6 +157,22 @@ def test_all_krakens(wait=False):
...
+       os.path.getctime(temp_target) > os.path.getctime(plain_target):
+        swap_temp = os.path.join(os.path.dirname(temp_target, 'x'))
+        os.rename(plain_target, swap_temp)
```
Comment (antoine-de) [inline fabfile/component/kraken.py]: same here, you should use fabtool.move

File: fabfile/instance.py [hunk 3]
```diff
@@ -74,8 +74,16 @@ def __init__(self, name, db_password, db_local='fr_FR.UTF8',
...
     @property
-    def target_lz4_file(self):
+    def plain_target_lz4_file(self):
         return "{base_dest}/{instance}/data.nav.lz4".format(base_dest=env.tyr_base_destination_dir, instance=self.name)
+    target_lz4_file = plain_target_lz4_file
+
+    @property
+    def temp_target_lz4_file(self):
+        target_path = os.path.join(env.tyr_base_destination_dir, self.name, 'temp')
+        if not os.path.exists(target_path):
```
Comment (antoine-de) [inline fabfile/instance.py]: same here, you should use fabtools

File: fabfile/component/kraken.py [hunk 4]
```diff
@@ -156,6 +157,22 @@ def test_all_krakens(wait=False):
...
+        swap_data_nav(instance)
+
+def swap_data_nav(instance):
+    """ swap old/new data.nav.lz4, only if new is still in temp directory
+    """
+    temp_target = instance.temp_target_lz4_file
+    plain_target = instance.plain_target_lz4_file
+    if files.is_file(temp_target) and files.is_file(plain_target) and \
+       files.getmtime(temp_target) > files.getmtime(plain_target):
+        swap_temp = os.path.join(os.path.dirname(temp_target, 'x'))
```
Comment (antoine-de) [inline fabfile/component/kraken.py]: what is the 'x' used for ?

File: fabfile/component/kraken.py [hunk 5]
```diff
@@ -156,6 +157,22 @@ def test_all_krakens(wait=False):
...
+        files.move(plain_target, swap_temp)
+        files.move(temp_target, plain_target)
+        files.move(swap_temp, temp_target)
```
Comment (antoine-de) [inline fabfile/component/kraken.py]: is it usefull to swap the file ?
I'm not sure to understand the workflow

File: fabfile/instance.py [hunk 6]
```diff
@@ -74,8 +75,16 @@ def __init__(self, name, db_password, db_local='fr_FR.UTF8',
...
-    def target_lz4_file(self):
+    def plain_target_lz4_file(self):
         return "{base_dest}/{instance}/data.nav.lz4".format(base_dest=env.tyr_base_destination_dir, instance=self.name)
+    target_lz4_file = plain_target_lz4_file
+
+    @property
+    def temp_target_lz4_file(self):
+        target_path = os.path.join(env.tyr_base_destination_dir, self.name, 'temp')
+        if not files.is_dir(target_path):
+            run("mkdir -p {}".format(target_path))
```
Comment (antoine-de) [inline fabfile/instance.py]: can't you use a fabtool require here ?

File: fabfile/tasks.py [hunk 7]
```diff
@@ -103,9 +103,11 @@ def upgrade_all(bina=True, up_tyr=True, up_confs=True, kraken_wait=True):
     with utils.send_mail():
         execute(check_last_dataset)
         if up_tyr:
-            execute(upgrade_tyr, up_confs=up_confs)
+            execute(upgrade_tyr, up_confs=up_confs and not bina)
```
Comment (antoine-de) [inline fabfile/tasks.py]: why don't you upgrade the configuration if there is `bina` ?

File: fabfile/component/kraken.py (hunk 4 above)
Comment (francois-vincent) [inline fabfile/component/kraken.py]: it's a temporary file

File: fabfile/component/kraken.py (hunk 5 above)
Comment (francois-vincent) [inline fabfile/component/kraken.py]: let's talk about it :)

File: fabfile/instance.py (hunk 6 above)
Comment (francois-vincent) [inline fabfile/instance.py]: this does not make a difference...

**Pull request merged**